### ISSUE TITLE
pipewire compatibility

### ISSUE THREAD

**Issue by bencebecsy** [lyrebird-voice-changer/lyrebird#107]
I'm not sure if this is related to other problems people were seeing, but I was trying to make `lyrebird` work on Pop-os (ubuntu based distro). I ran into the problem of not seeing the lyrebird output as an option in `pavucontrol`, and getting the error message:
```
No PulseAudio daemon running, or not running as session daemon.
```
I think that is because Pop-os as some other distros switched to using `pipewire` instead of `pulseaudio`, so `pacmd` is not fully supported. When I tried to run
```
pacmd load-module module-null-sink sink_name=Lyrebird-Output
```
I got the same error as above. However, I found this page (https://gitlab.freedesktop.org/pipewire/pipewire/-/wikis/Migrate-PulseAudio#module-null-sink) to find the equivalent command in `pipewire` to be:
```
pw-cli create-node adapter factory.name=support.null-audio-sink media.class="Audio/Sink" object.linger=1 monitor.channel-volumes=1 node.name=Lyrebird-Output
```
Running this while `lyrebird` is running seems to solve my issue as it creates that sink names Lyrebird-Output. I'm not sure how hard it would be to implement this change in `lyrebird`, but I thought I'd share this as a hacky workaround and a potential starting point for a future `pipewire` integration.

**Comment by orlandohuang** [comment]
I straight up can't even install lyrebird...
```
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Note, selecting 'lyrebird' instead of '/home/<myusername>/Downloads/lyrebird_1.1.0-3.deb'
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 lyrebird : Depends: pulseaudio
E: Unable to correct problems, you have held broken packages.
```

**Comment by bencebecsy** [comment]
Hmm, interesting. I had problems with the package too, but I managed to install from the tar.gz with the install script.

**Comment by develroo** [comment]
Ok just to add to this.. I tried the fork and it would not work. But The original deb package does. I too am on Pipewire so I am not sure what the fork does differently but the output is a clue.



```
lyrebird 
No PulseAudio daemon running, or not running as session daemon.
No PulseAudio daemon running, or not running as session daemon.
No PulseAudio daemon running, or not running as session daemon.
Traceback (most recent call last):
  File "/usr/local/share/lyrebird/app/mainwindow.py", line 234, in toggle_activated
    null_sink = subprocess.check_call(
  File "/usr/lib/python3.10/subprocess.py", line 369, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['pacmd', 'load-module', 'module-null-sink', 'sink_name=Lyrebird-Output']' returned non-zero exit status 1.
Traceback (most recent call last):
  File "/usr/local/share/lyrebird/app/mainwindow.py", line 280, in preset_clicked
    self.set_preset(preset)
  File "/usr/local/share/lyrebird/app/mainwindow.py", line 216, in set_preset
    self.restart_sox()
  File "/usr/local/share/lyrebird/app/mainwindow.py", line 197, in restart_sox
    command = utils.build_sox_command(
  File "/usr/local/share/lyrebird/app/core/utils.py", line 24, in build_sox_command
    params_str = ' '.join(map(str, effect_params))
TypeError: 'bool' object is not iterable
No PulseAudio daemon running, or not running as session daemon.
No PulseAudio daemon running, or not running as session daemon.

```
The deb package just gives this but works still.


```
lyrebird 
No PulseAudio daemon running, or not running as session daemon.
No PulseAudio daemon running, or not running as session daemon.
No PulseAudio daemon running, or not running as session daemon.
No PulseAudio daemon running, or not running as session daemon.

```

Maybe some regression testing is required?

**Comment by doubledge-spec** [comment]
Same issue on Arch
````Traceback (most recent call last):
  File "/usr/share/lyrebird/app.py", line 13, in <module>
    win = mainwindow.MainWindow()
  File "/usr/share/lyrebird/app/mainwindow.py", line 62, in __init__
    utils.unload_pa_modules(check_state=False)
  File "/usr/share/lyrebird/app/core/utils.py", line 60, in unload_pa_modules
    subprocess.call('pacmd unload-module module-null-sink'.split(" "))
  File "/usr/lib/python3.10/subprocess.py", line 345, in call
    with Popen(*popenargs, **kwargs) as p:
  File "/usr/lib/python3.10/subprocess.py", line 969, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/usr/lib/python3.10/subprocess.py", line 1845, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'pacmd'```

**Comment by linuxgnuru** [comment]
yeah, most distros now use pipewire so this will need to be rewritten

**Comment by ssieb** [comment]
But pipewire has pulseaudio compatibility, so it should work anyway.

**Comment by LuNeder** [comment]
+1 to this, would be really great to have PipeWire compatibility

**Comment by Steccah** [comment]
[This](https://github.com/lyrebird-voice-changer/lyrebird/tree/pulse-unload-fix) kinda works, there is a huge delay

**Comment by harrego** [comment]
Merged with #62
